Vagrantfile

```diff
@@ -319,23 +321,41 @@ Vagrant.configure("2") do |config|
         host.hostsupdater.aliases = info[:alias]
       end
 
-      host.vm.provider :virtualbox do |vb|
-        vb.customize ["modifyvm", :id, "--cpus", info[:cpus]]
-        vb.customize ["modifyvm", :id, "--hwvirtex", "on"]
-        vb.customize ["modifyvm", :id, "--memory", info[:mem]]
-        vb.customize ["modifyvm", :id, "--name", hostname]
-        vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
-        vb.customize ["modifyvm", :id, "--natdnshostresolver2", "on"]
-        vb.gui = info[:show_gui]
-        vb.name = hostname
+      if(vagrant_provider == 'virtualbox')
+        host.vm.provider :virtualbox do |vb|
+          vb.customize ["modifyvm", :id, "--cpus", info[:cpus]]
+          vb.customize ["modifyvm", :id, "--hwvirtex", "on"]
+          vb.customize ["modifyvm", :id, "--memory", info[:mem]]
+          vb.customize ["modifyvm", :id, "--name", hostname]
+          vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
+          vb.customize ["modifyvm", :id, "--natdnshostresolver2", "on"]
+          vb.gui = info[:show_gui]
+          vb.name = hostname
+        end
+      elsif(vagrant_provider == 'libvirt')
+        host.vm.provider :libvirt do |libvirt|
+          libvirt.cpus = info[:cpus]
+          libvirt.memory = info[:mem]
+          libvirt.nested = true
+        end
       end
-
       host.vm.hostname = hostname
       if(hostname.include? base_box_builder_vm_name)
         host.vm.provision "shell" do |s|
           s.path = "scripts/linux/install-docker.sh"
           s.args = ["--user", "vagrant"]
         end
+        
+        config.ssh.insert_key = false
+        
+        if(vagrant_provider == 'libvirt')
+          $enableSshPasswordAuthentication = <<-'SCRIPT'
+          sudo grep -q "^PasswordAuthentication" /etc/ssh/sshd_config && sudo sed -i "s/^PasswordAuthentication no/PasswordAuthentication yes/" /etc/ssh/sshd_config || sudo sed -i -e "\$aPasswordAuthentication yes" /etc/ssh/sshd_config
```

Comment:
Is `sudo` needed here? The shell provisioner runs scripts as privileged users [by default](https://www.vagrantup.com/docs/provisioning/shell.html#privileged)